### ISSUE TITLE
Removing linear regression support in Statsample

### ISSUE THREAD

**Issue by lokeshh** [SciRuby/statsample#53]
I think its reasonable to remove linear regression support in Statsample because of following reasons:

- Its equivalent which is Normal regression lies in Statsample-GLM
- This way all regression models would lie in Statsample-GLM
- Statsample-GLM already supports predicting on new data which Statsample doesn't yet
- Normal regression in Statsample-GLM works without an intercept providing more flexibility
- This will keep regression models management limited to Statsample-GLM and Statsample gem would be responsible for other statistic tools other than regression, hence simplifying the management.

**Comment by v0dro** [comment]
Works. Go ahead.

**Comment by lokeshh** [comment]
@v0dro Ok. Well, then it also implies https://github.com/SciRuby/statsample/pull/51 would be no longer needed. Right?

**Comment by v0dro** [comment]
Are there any particular cases in statsample that are not covered in statsample-glm? If no, then you can safely remove linear regression from statsample and not re-implement the formula language in this  gem too (I think that's what you're doing given the commits in #51 - I've been out on vacation so could not follow up on daily activities).

**Comment by lokeshh** [comment]
Yes you are right. I was implementing formula language in #51. I'm removing it now and will now add a PR to remove multiple regression.

**Comment by lokeshh** [comment]
Statsample is much faster than Statsample-GLM as [these benchmark](https://gist.github.com/lokeshh/8fed775e53ee12c296e9fd3d39f6d8ea) shows.
